src/core-api.client.ts

```diff
@@ -182,12 +177,12 @@ export class CoreAPIClient {
    * @param coreSQL
    * @param dtoNames
    */
-  public async query<T extends { [projection: string]: DTOModels }>(coreSQL: string, dtoNames: DTOName[]): Promise<{ data: T[] }> {
+  public async query<T extends { [projection: string]: DTOModels }>(coreSQL: string, dtoNames?: DTOName[]): Promise<{ data: T[] }> {
     const token = await this._ensureToken();
-    const params = new URLSearchParams({
+    const params = new URLSearchParams(Object.assign({ 
       ...RequestOptionsFacory._getRequestAccountQueryParams(token, this._config),
-      dtos: RequestOptionsFacory.getDTOVersionsString(dtoNames)
-    });
+      dtos: RequestOptionsFacory.getDTOVersionsString(dtoNames || [... new Set(coreSQL.match(/(?<=from\s*|join\s*)(\w+)/gi))] as DTOName[]),
```

Comment:
There are possible cases where this fails with grouping and sub in queries.
For usability, we should keep this explicit on the public API of this client - to keep the magic out and easy understandable. 
`DTOName` type helps in the IDE with auto complete also the dirty type cast then is not needed.
You can of cause still do regexing when calling the client externally. Please revert.

Reply:
Some possible cases examples would be nice to see. Maybe they can be addressed by adjusting the regex.